src/Analysis/GWASV2/Steps/SelectCovariates/SelectCovariates.jsx

```diff
@@ -2,49 +2,62 @@ import React, { useState } from 'react';
 import { PropTypes } from 'prop-types';
 import ContinuousCovariates from '../../Components/Covariates/ContinuousCovariates';
 import CustomDichotomousCovariates from '../../Components/Covariates/CustomDichotomousCovariates';
+import CovariatesCardsList from '../../Components/Covariates/CovariatesCardsList';
+import ACTIONS from '../../Shared/StateManagement/Actions';
 
 const SelectCovariates = ({
   dispatch,
   studyPopulationCohort,
   outcome,
   covariates,
 }) => {
-  const [mode, setMode] = useState(undefined);
-  const [selectedCovariate, setSelectedCovariate] = useState({});
+  const [selectionMode, setSelectionMode] = useState('');
 
   return (
     <React.Fragment>
-      {mode === 'continuous' && (
+      {selectionMode === 'continuous' && (
         <ContinuousCovariates
-          setMode={setMode}
-          selected={selectedCovariate}
-          dispatch={dispatch}
-          handleSelect={setSelectedCovariate}
           covariates={covariates}
           outcome={outcome}
-          type={'covariate'}
+          handleClose={() => {
+            setSelectionMode('');
+          }}
+          dispatch={(chosenOutcome) => {
```

Comment:
consider renaming to `chosenCovariate`

Reply:
Good catch! Thank you! This has been updated to ```chosenCovariate```